modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/mutate/Mutate.scala

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (c) 2014-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.iglu.schemaddl.jsonschema.mutate
+
+import com.snowplowanalytics.iglu.schemaddl.jsonschema.Schema
+import com.snowplowanalytics.iglu.schemaddl.jsonschema.properties._
+
+object Mutate {
+
+  /**
+  * Mutates a schema to be more amenable for converting into a storage AST.
+  *
+  * Some JsonSchema properties are helpful for strict validation, but have no meaning when storing
+  * the data in a strictly typed storage warehouse. It is helpful to remove these jsonschema features:
+  * - oneOf / anyOf properties
+  * - additionalProperties of an object
+  * - an array represented as a tuple of different types
+  * 
+  * _Most_ data that validates against the input schema will also validate against the output schema.
```

Comment:
I'm not sure if it would be useful/possible, but what about creating some unit tests to prove this statement? We could also create tests for cautionary examples to show that it fails.

Reply:
It's a great idea!  But honestly I can't think how to write such a test in any meaningful way.

I could write a completely trivial test with a couple of test schemas and some input data.  But I don't think it's any better than the tests I've already written.

It would be awesome if we could use property-based testing to generate random schemas and random data to test the assertion.  But... random data is unlikely to ever validate against a random schema 😉